kubespawner/spawner.py

```diff
@@ -1895,7 +1896,6 @@ def stop(self, now=False):
         except TimeoutError:
             self.log.error("Pod %s did not disappear, restarting pod reflector", self.pod_name)
             self._start_watching_pods(replace=True)
```

Comment:
```suggestion
            self._start_watching_pods(replace=True)
            raise
```

This is to revert a change that I think is problematic. By doing so, we can get this test merged.